doc/nrf/libraries/others/emds.rst

```diff
@@ -14,43 +14,55 @@ Implementation
 **************
 The :kconfig:option:`CONFIG_EMDS` Kconfig option enables the emergency data storage.
 
-The application must initialize the pre-allocated storage area by using the :c:func:`emds_init` function.
-The :kconfig:option:`CONFIG_EMDS_SECTOR_COUNT` option defines how many sectors should be used to store data.
+The EMDS is implemented to use the persistent memory in form of storage partitions, which can be either flash or RRAM.
+The static partition manager file :file:`pm.yml.emds` is required for defining the storage partitions.
+The storage partitions should be aligned to write block size, which is 4 bytes for flash and 16 bytes for RRAM.
+The EMDS mandates at least two storage partitions to allow the fallback mechanism.
+The application must initialize storage partitions by using the :c:func:`emds_init` function.
 
-The allocated storage space must be larger than the combined data stored by the application.
+The partition storage space must be larger than the combined data stored by the application.
 Allocating a larger storage area will demand more resources, but also increase the life expectancy of the persistent memory.
 The chosen size should reflect the amount of data stored, the available persistent memory and how the application calls the :c:func:`emds_store` function.
-In general, it should not be necessary to allocate a large storage area, since only a limited set of data should be stored to ensure swift completion of writing the data on shutdown.
 
 The memory location that is going to be stored must be added on initialization.
 All memory areas must be provided through entries containing an ID, data pointer, and data length, using the :c:func:`emds_entry_add` function and the :c:macro:`EMDS_STATIC_ENTRY_DEFINE` macro.
 Entries to be stored when the emergency data storage is triggered need their own unique IDs that are not changed after a reboot.
-
-When all entries are added, the :c:func:`emds_load` function restores the entries into the memory areas from the persistent memory.
+The entries are stored in the partition in form of snapshots.
+The snapshot is a packed copy of all entries that are registered in the EMDS stored in the data area of the partition.
+Additionally, each snapshot is described by the snapshot metadata that is stored in the metadata area of the partition.
+
+When all entries are added, the :c:func:`emds_load` function starts a partition scanning procedure to find the latest snapshot.
+The EMDS scans the metadata area of both partitions.
+The :kconfig:option:`CONFIG_EMDS_SCANNING_FAILURES` option configures the depth of the scanning procedure.
+If the scanning procedure fails the configured number of times in the row, the scanning is stopped, since highly likely there is no stored metadata anymore.
```

Comment:
```suggestion
If the scanning procedure fails the configured number of times in a row, the scanning is stopped, because most likely, there is no stored metadata left.
```